syncs/lock_example_test.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package syncs_test
+
+import (
+	"encoding/hex"
+	"log"
+	"sync"
+
+	"tailscale.com/syncs"
+)
+
+func ExampleLockFunc() {
+	var nodesMu sync.Mutex
+	var nodes []string
+	syncs.LockFunc(&nodesMu, func() { nodes = append(nodes, "node123") })
+}
+
+func ExampleLockValue() {
+	var nodesMu sync.Mutex
+	var nodes []string
+	n := syncs.LockValue(&nodesMu, func() int { return len(nodes) })
+	log.Printf("there are %d nodes", n)
```

Comment:
prefer fmt.Printf/printf, and then you can add `// Output: ` lines which will be used to make assertions that the output is still stable - making these functional tests